slither/tools/read_storage/read_storage.py

```diff
@@ -111,9 +129,46 @@ def get_storage_layout(self) -> None:
                 elif isinstance(type_, ArrayType):
                     elems = self._all_array_slots(var, contract, type_, info.slot)
                     tmp[var.name].elems = elems
-
+        tmp = self.get_unstructured_layout(tmp)
         self._slot_info = tmp
 
+    def get_unstructured_layout(self, tmp: dict) -> dict:
+        for _, var in self.constant_slots:
+            var_name = var.name
+            try:
+                exp = var.expression
+                if isinstance(exp, CallExpression) and str(exp.called) == "keccak256(bytes)":
+                    exp = exp.arguments[0]
```

Comment:
I didn't think ConstantFolding would accept a CallExpression, since we have
```python
CONSTANT_TYPES_OPERATIONS = Union[
    Literal, BinaryOperation, UnaryOperation, Identifier, TupleExpression, TypeConversion
]
```
So these two lines extract the inner expression, which can then fall through to the following lines where ConstantFolding is used.